Design new .NET Core Setup/Install area; Rewrite prereqs articles; Add package managers (#15569)
dotnet/docs: docs/core/setup/dependencies.md

---

> title: Core SDK and runtime dependencies for Windows, Linux, and macOS - .NET Core

Reviewer: ```suggestion
title: .NET Core SDK and runtime dependencies for Windows, Linux, and macOS - .NET Core
```

Reviewer: should the word requirement be on the title metadata too?

> description: Details the operating system and CPU architecture prerequisites to install the .NET Core SDK and runtime on on Windows, Linux, and macOS.

Reviewer: ```suggestion
description: Details the operating system and CPU architecture prerequisites to install the .NET Core SDK and runtime on Windows, Linux, and macOS.
```

author: leecow
ms.author: leecow
ms.date: 11/06/2019
zone_pivot_groups: operating-systems-set-one
---

# .NET Core dependencies and requirements

> This article details which operating systems and CPU architecture are supported by .NET Core.

Reviewer: Well the right acronym for operating systems would be OSs. I just don't use that because it can be easily confused with OSS.

## Supported operating systems

::: zone pivot="os-windows"

<!-- markdownlint-disable MD025 -->
<!-- markdownlint-disable MD024 -->

# [.NET Core 3.0](#tab/netcore30)

The following Windows operating systems are supported with .NET Core 3.0:

> [!NOTE]
> A `+` symbol represents the minimum version.

| OS                            | Version                        | Architectures   |
| ----------------------------- | ------------------------------ | --------------- |
| Windows Client                | 7 SP1+, 8.1                    | x64, x86        |
| Windows 10 Client             | Version 1607+                  | x64, x86        |
| Windows Server                | 2012 R2+                       | x64, x86        |
| Nano Server                   | Version 1803+                  | x64, ARM32      |

> For more information about .NET Core 3.0 supported OS, distributions, and lifecycle policy, see [.NET Core 3.0 Supported OS Versions](https://github.com/dotnet/core/blob/master/release-notes/3.0/3.0-supported-os.md).

Reviewer: OS is singular. OSes would be the plural which I don't like. This shows up more than one time in this PR.

```suggestion
For more information about .NET Core 3.0 supported operating systems, distributions, and lifecycle policy, see [.NET Core 3.0 Supported OS Versions](https://github.com/dotnet/core/blob/master/release-notes/3.0/3.0-supported-os.md).
```

# [.NET Core 2.2](#tab/netcore22)

> The following Windows operating systems are supported with .NET Core 2.2:

Reviewer: Should it be versions? It happens more than once as well.

```suggestion
The following Windows versions are supported with .NET Core 2.2:
```

For more information about .NET Core 2.2 supported OS, distributions, and lifecycle policy, see [.NET Core 2.2 Supported OS Versions](https://github.com/dotnet/core/blob/master/release-notes/2.2/2.2-supported-os.md).

# [.NET Core 2.1](#tab/netcore21)

The following Windows operating systems are supported with .NET Core 2.1:

> [!NOTE]
> A `+` symbol represents the minimum version.

| OS                            | Version                        | Architectures   |
| ----------------------------- | ------------------------------ | --------------- |
| Windows Client                | 7 SP1+, 8.1                    | x64, x86        |
| Windows 10 Client             | Version 1607+                  | x64, x86        |
| Windows Server                | 2008 R2 SP1+                   | x64, x86        |
| Nano Server                   | Version 1803+                  | x64,            |

For more information about .NET Core 2.1 supported OS, distributions, and lifecycle policy, see [.NET Core 2.1 Supported OS Versions](https://github.com/dotnet/core/blob/master/release-notes/2.1/2.1-supported-os.md).

---

<!-- markdownlint-disable MD001 -->

### Windows 7 / 8.1 / Server 2008 R2

> Additional dependencies are required if you're installing the .NET SDK or Runtime on the following Windows operating systems:

Reviewer: ```suggestion
Additional dependencies are required if you're installing the .NET SDK or runtime on the following Windows operating systems:
```

- Windows 7 SP1
- Windows 8.1
- Windows Server 2008 R2
- Windows Server 2012 R2

Install the following:

> - [Microsoft Visual C++ 2015 Redistributable Update 3](https://www.microsoft.com/download/details.aspx?id=52685).

Reviewer: wasn't this requirement needed for any version when those conditions were met?

![image](https://user-images.githubusercontent.com/12971179/69450960-271d6900-0d13-11ea-891a-54998f8265b8.png)

Author: According to the download page for these patches, they dont apply to windows 10 or the latest server releases. Regardless of how you install/run/whatever, on the windows versions listed, you need the patches.

> - [KB2533623](https://support.microsoft.com/help/2533623/microsoft-security-advisory-insecure-library-loading-could-allow-remot)

Reviewer: You don't need this on Windows 8.1 and 2012

Also, where is KB2999226? That one was required for all 4.

Author: KB2999226 is included in the C++ redistributable.

---

## Linux distribution dependencies

Based on your linux distribution, you may need to install additional dependencies.

> [!IMPORTANT]
> The exact versions and names may vary slightly on your Linux distribution of choice.

### Ubuntu

Ubuntu distributions require the following libraries installed:

- liblttng-ust0
- libcurl3 (for 14.x and 16.x)
- libcurl4 (for 18.x)
- libssl1.0.0
- libkrb5-3
- zlib1g
- libicu52 (for 14.x)
- libicu55 (for 16.x)
- libicu57 (for 17.x)
- libicu60 (for 18.x)

For .NET Core apps that use the *System.Drawing.Common* assembly, you also need the following dependency:

- libgdiplus (version 6.0.1 or later)

> [!WARNING]
> Most versions of Ubuntu include an earlier version of libgdiplus. You can install a recent version
> of libgdiplus by adding the Mono repository to your system. For more information,
> see <https://www.mono-project.com/download/stable/>.

### CentOS and Fedora

CentOS distributions require the following libraries installed:

- lttng-ust
- libcurl
- openssl-libs
- krb5-libs
- libicu
- zlib

> Fedora users: If your openssl's version >= 1.1, you'll need to install **compat-openssl10**.

Reviewer: ```suggestion
Fedora users: If your OpenSSL's version >= 1.1, you'll need to install **compat-openssl10**.
```

> For .NET Core 2.0, following dependencies are also required:

Reviewer: ```suggestion
For .NET Core 2.0, the following dependencies are also required:
```

::: zone-end

::: zone pivot="os-macos"

.NET Core is supported on the following macOS releases:

> [!NOTE]
> A `+` symbol represents the minimum version.

| .NET Core Version | macOS                 | Architectures |     |
| ----------------- | --------------------- | --------------| --- |
| 3.0               | High Sierra (10.13+)  | x64 | [More information](https://github.com/dotnet/core/blob/master/release-notes/3.0/3.0-supported-os.md) |
| 2.2               | Sierra (10.12+)       | x64 | [More information](https://github.com/dotnet/core/blob/master/release-notes/2.2/2.2-supported-os.md) |
| 2.1               | Sierra (10.12+)       | x64 | [More information](https://github.com/dotnet/core/blob/master/release-notes/2.1/2.1-supported-os.md) |

## libgdiplus

.NET Core applications that use the *System.Drawing.Common* assembly require libgdiplus to be installed.

An easy way to obtain libgdiplus is by using the [Homebrew ("brew")](https://brew.sh/) package manager for macOS. After installing *brew*, install libgdiplus by executing the following commands at a Terminal (command) prompt:

```console
brew update
brew install mono-libgdiplus
```

::: zone-end

## Next steps

> - To develop and run apps, install the [.NET Core SDK](install-sdk.md). (includes the runtime)

Reviewer: ```suggestion
- To develop and run apps, install the [.NET Core SDK](install-sdk.md) (includes the runtime).
```

Reviewer 2: Should we use the "Next steps" button for any next steps links? Like this: https://docs.microsoft.com/en-us/visualstudio/get-started/csharp/tutorial-console?view=vs-2019#next-steps

> [!div class="nextstepaction"]
> [Install .NET Core runtime](install-runtime.md)

> - To run apps others have created, install the [.NET Core runtime](install-runtime.md).

Reviewer: what if it's self-contained?

should you say "you might need to install"?
